alvr/server/cpp/platform/linux/EncodePipelineSW.cpp

```diff
@@ -54,39 +64,47 @@ alvr::EncodePipelineSW::EncodePipelineSW(VkFrame &input_frame, VkFrameCtx& vk_fr
   switch (codec_id)
   {
     case ALVR_CODEC_H264:
-      encoder_ctx->profile = settings.m_use10bitEncoder ? FF_PROFILE_H264_HIGH_10 : FF_PROFILE_H264_HIGH;
-      AVUTIL.av_dict_set(&opt, "preset", "ultrafast", 0);
-      AVUTIL.av_dict_set(&opt, "tune", "zerolatency", 0);
-      encoder_ctx->gop_size = 72;
+      encoder_ctx->profile = FF_PROFILE_H264_HIGH;
       break;
     case ALVR_CODEC_H265:
       encoder_ctx->profile = settings.m_use10bitEncoder ? FF_PROFILE_HEVC_MAIN_10 : FF_PROFILE_HEVC_MAIN;
-      AVUTIL.av_dict_set(&opt, "preset", "ultrafast", 0);
-      AVUTIL.av_dict_set(&opt, "tune", "zerolatency", 0);
-      encoder_ctx->gop_size = 72;
       break;
   }
 
+  switch (Settings::Instance().m_rateControlMode)
+  {
+    case ALVR_CBR:
+      av_dict_set(&opt, "nal-hrd", "cbr", 0);
+      break;
+    case ALVR_VBR:
+      av_dict_set(&opt, "nal-hrd", "vbr", 0);
+      break;
+  }
+
+  AVUTIL.av_dict_set(&opt, "preset", "ultrafast", 0);
+  AVUTIL.av_dict_set(&opt, "tune", "zerolatency", 0);
 
   encoder_ctx->width = width;
   encoder_ctx->height = height;
   encoder_ctx->time_base = {1, (int)1e9};
   encoder_ctx->framerate = AVRational{settings.m_refreshRate, 1};
   encoder_ctx->sample_aspect_ratio = AVRational{1, 1};
-  encoder_ctx->pix_fmt = settings.m_use10bitEncoder ? AV_PIX_FMT_YUV420P10LE : AV_PIX_FMT_YUV420P;
+  encoder_ctx->pix_fmt = settings.m_use10bitEncoder && codec_id == ALVR_CODEC_H265 ? AV_PIX_FMT_YUV420P10 : AV_PIX_FMT_YUV420P;
   encoder_ctx->max_b_frames = 0;
+  encoder_ctx->gop_size = 0;
```

Comment:
Just in case, while it's fixing the issue, it's producing only IDR frames when set to 0.